qpid-proton-cpp/reconnect/direct-rcv-close-link.cpp

```diff
@@ -0,0 +1,89 @@
+/*
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ *
+ */
+
+#include <proton/container.hpp>
+#include <proton/delivery.hpp>
+#include <proton/message.hpp>
+#include <proton/message_id.hpp>
+#include <proton/messaging_handler.hpp>
+#include <proton/listen_handler.hpp>
+#include <proton/listener.hpp>
+
+#include <iostream>
+
+
+class direct_recv : public proton::messaging_handler {
+  private:
+    class listener_ready_handler : public proton::listen_handler {
+        void on_open(proton::listener& l) override {
+            std::cout << "listening on " << l.port() << std::endl;
+        }
+    };
+
+    std::string url;
```

Comment:
This file in general is not observing the conventions of the Equipage C++ code.  See https://github.com/amqphub/equipage/blob/main/qpid-proton-cpp/receive.cpp and others and note the trailing underscore for member vars, dlv and rcv and conn and so on for commonly function vars.